### ISSUE TITLE
InvalidOperationException thrown when calling OnItemChecked in class derived from ListView (in VirtualMode)

### ISSUE THREAD

**Issue by djoyner-aveva** [dotnet/winforms#11658]
### .NET version

.NET 8

### Did it work in .NET Framework?

Yes

### Issue description

We have a class derived from ListView. It calls OnItemChecked in order to raise the ItemChecked event to subscribers. In the ItemCheckedEventArgs we pass in a cloned ListViewItem so that we can set the Checked property correctly. This worked fine in .NET Framework as OnItemChecked just raised the event. However, in .NET 8, it's throwing an InvalidOperationException:
"Cannot get accessiblity object when ListViewItem is not attached to a ListView."
An accessibility object has been created for the ListView (which seems reasonable), however, the cloned ListViewItem is not hosted by the ListView, which I think is the cause of the exception.

### Steps to reproduce

Derive a class from ListView
Ensure VirtualMode and Checkboxes are enabled
Ensure the accessibility object is created (not sure how to force this?)
Add a method that calls OnItemChecked - something like:

```
ListViewItem lvi = HitTest(...); // Any code that gets a ListViewItem from the ListView
var update = (ListViewItem) lvi.Clone();
update.Checked = true;
Debug.Assert(update.Checked);
Debug.Assert(update.ListView == null);
OnItemChecked(new ItemCheckedEventArgs(update)); // throws exception
```

**Comment by elachlan** [comment]
@djoyner-aveva Can you please provide a stack trace?

**Comment by elachlan** [comment]
@Olina-Zhang can your team please test this?

**Comment by Zheng-Li01** [comment]
@djoyner-aveva , created the project based on your description to reproduce the issue, no exception pops up as below screenshot, could you please check my reproduce project and correct it if there missing any steps.
[WinFormsApp25.zip](https://github.com/user-attachments/files/16175422/WinFormsApp25.zip)
![GH11658](https://github.com/dotnet/winforms/assets/38325459/a58117a3-e3f2-4e71-89a6-0a89a0de0a15)

**Comment by djoyner-aveva** [comment]
[WinFormsApp3.zip](https://github.com/user-attachments/files/16175845/WinFormsApp3.zip)

Modified the sample slightly to add the control to the form, and this reproduces the error

**Comment by djoyner-aveva** [comment]
Call stack from sample app is:
System.InvalidOperationException: Cannot get accessiblity object when ListViewItem is not attached to a ListView.
   at System.Windows.Forms.ListViewItem.get_AccessibilityObject()
   at System.Windows.Forms.ListView.OnItemChecked(ItemCheckedEventArgs e)
   at WinFormsApp3.ListViewTest.TryCrash() in D:\Test Projects\WinFormsApp3 - Copy (2)\WinFormsApp3\ListViewTest.cs:line 30
   at WinFormsApp3.Form1.button1_Click(Object sender, EventArgs e) in D:\Test Projects\WinFormsApp3 - Copy (2)\WinFormsApp3\Form1.cs:line 12
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)

**Comment by Zheng-Li01** [comment]
@djoyner-aveva, thanks for your project, cannot reproduce it on both .NET Framework and .NET that no exceptions pops up after clicked the button as below screenshot, could you please have a check?
![GHRT11658](https://github.com/user-attachments/assets/b34c89ad-63b7-48bf-9b11-c568e97794b1)

**Comment by elachlan** [comment]
@Zheng-Li01, you need to trigger the Accessibility Object via Accessibility insights for windows (I think).

**Comment by Zheng-Li01** [comment]
@elachlan  thanks for your guidence, yes, the issue can reproduce from .NET 6.0 ~ 9.0 via Accessibility insights for windows as below screenshot.
![image](https://github.com/user-attachments/assets/370d41a5-c42c-4cf8-b129-6686f4664b97)

**Comment by elachlan** [comment]
@djoyner-aveva Just so I understand the use case clearly, why are you cloning the `ListViewItem`? 

We can change the `OnItemChecked` to only trigger the AccessibilityObject change when the item exists in the Items collection, but it seems unusual to pass an item in that doesn't exist in the collection.

```csharp
protected virtual void OnItemChecked(ItemCheckedEventArgs e)
{
    _onItemChecked?.Invoke(this, e);

    if (!CheckBoxes)
    {
        return;
    }

    if (IsAccessibilityObjectCreated && Items.Contains(e.Item))
    {
        ListViewItem item = e.Item;
        ToggleState oldValue = item.Checked ? ToggleState.ToggleState_Off : ToggleState.ToggleState_On;
        ToggleState newValue = item.Checked ? ToggleState.ToggleState_On : ToggleState.ToggleState_Off;
        item.AccessibilityObject.RaiseAutomationPropertyChangedEvent(UIA_PROPERTY_ID.UIA_ToggleToggleStatePropertyId, (VARIANT)(int)oldValue, (VARIANT)(int)newValue);
    }
}
```

**Comment by djoyner-aveva** [comment]
@elachlan 
Thanks for your question. We're cloning it because writing to Checked on the original ListViewItem doesn't work:

```
item.Checked = true;
Debug.Assert(item.Checked); // fails
```
We have the data for the checked states stored outside of the ListView and we want to update them in response to an
 ItemChecked event. I've updated the sample app to try and demonstrate this (hopefully it's a bit more meaningful).

[WinFormsApp3.zip](https://github.com/user-attachments/files/16190093/WinFormsApp3.zip)

**Comment by djoyner-aveva** [comment]
@elachlan 
And I like the look of your second suggestion - that was the fix I had in mind

`if (IsAccessibilityObjectCreated && e.Item.ListView == this)`

**Comment by lonitra** [comment]
@SimonZhao888 Could you confirm that this does not repro in Framework?

**Comment by Tanya-Solyanik** [comment]
@SimonZhao888 - please add a manual test case that clones a ListView item to the WinformsControlsApp

**Comment by SimonZhao888** [comment]
> @SimonZhao888 Could you confirm that this does not repro in Framework?

@lonitra , this does not repro in Framework.

**Comment by LeafShi1** [comment]
> @SimonZhao888 Could you confirm that this does not repro in Framework?

@lonitra @Tanya-Solyanik 
Due to this issue appears in the [AccessibilityObject.RaiseAutomationPropertyChangedEvent](https://github.com/dotnet/winforms/blob/e4085eb08ceef38d44b0bcdae0bc02e374794979/src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListView/ListView.cs#L4843), But in .netFramwork, [AccessibilityObject is not checked](https://vstfdevdiv.corp.microsoft.com/DevDiv2/devdiv/_versionControl?path=%24/DevDiv/FX/Feature/NetFXDev1/NDP/fx/src/WinForms/Managed/System/WinForms/ListView.cs&version=T&line=4438&lineEnd=4443&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), so there is no problem.

The issue only occurs with cloned objects, so please help confirm if this is a real bug that we need to fix.

**Comment by Tanya-Solyanik** [comment]
@LeafShi1  - thank you for bringing this up, yes please fix it and add a unit test.

**Comment by MelonWang1** [comment]
Verified this issue in the latest .NET9.0 SDK build: 9.0.100-rc.2.24420.21 + dlls built from Winforms main branch, it was fixed. OnItemChecked method does not throw exception when the ListViewItem is clone object.

https://github.com/user-attachments/assets/1f549da2-a0e2-4cea-a204-d0617273f2cb

**Comment by Philip-Wang01** [comment]
Verified this issue in the latest .NET9.0 SDK build: 9.0.100-rc.2.24420.21 + dlls built from release/9.0 branch, it was fixed. The test results are the same as above.
